test/utils/fixture/fixture.go

```diff
@@ -101,6 +101,12 @@ func SetupWithOptions(name string, nrtList nrtv1alpha2.NodeResourceTopologyList,
 	}
 	ginkgo.By(fmt.Sprintf("set up the test namespace %q", ns.Name))
 
+	if len(nrtList.Items) == 0 {
+		gomega.Eventually(func(g gomega.Gomega) {
+			err = e2eclient.Client.List(context.TODO(), &nrtList)
+			g.Expect(err).ToNot(gomega.HaveOccurred())
+		}).WithTimeout(30*time.Second).WithPolling(10*time.Second).Should(gomega.Succeed(), "failed to pull NRT items")
+	}
```

Comment:
unfortunately this is not OK because the bug was to have the `ginkgo.By` in this function 2 lines above. We should not have assertions/expectations in this function. Actually, only functions prefixed with `Must` should have them. So let's convert this loop in a wait loop (example: `wait.PollUntilContextTimeout()`) and let's return error on failure

Reply:
If we fetch the nrtList before every suite, which I'd not be against, we should perhaps check the nrtList we just collected against the one we collected when everything started. If they differ, we should emit a fat big warning because this is not expected to happen. WDYT?

Reply:
Thanks for your comments. I'll reposition ginkgo.by few lines above to align with the ns setup step.
And thanks for sharing the info on functions starting with Must. I'll update this part and adopt your suggestion.

Reply:
addressed in last commit